internal/services/v1/relationships_test.go

```diff
@@ -982,6 +984,86 @@ func TestDeleteRelationships(t *testing.T) {
+		"document:base64YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==#owner@user:base64YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==": {},
+		"document:veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryverylong#owner@user:veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryverylong": {},
+	}
+
+	iterations := 0
+	for i := 0; i < 10; i++ {
+		iterations++
+
+		headRev, err := ds.HeadRevision(context.Background())
+		require.NoError(err)
+
+		beforeDelete := readOfType(require, "document", client, zedtoken.MustNewFromRevision(headRev))
+
+		resp, err := client.DeleteRelationships(context.Background(), &v1.DeleteRelationshipsRequest{
+			RelationshipFilter: &v1.RelationshipFilter{
+				ResourceType: "document",
+			},
+			OptionalLimit:                 5,
+			OptionalAllowPartialDeletions: true,
+		})
+		require.NoError(err)
+
+		headRev, err = ds.HeadRevision(context.Background())
+		require.NoError(err)
+
+		afterDelete := readOfType(require, "document", client, zedtoken.MustNewFromRevision(headRev))
+		require.LessOrEqual(len(beforeDelete)-len(afterDelete), 5)
```

Comment:
If this has 11 rels, it will be deleted in 3 iterations
- the first batch is tested with `i=0`
- the second batch is not tested, because it's partial, but `i != 0`
- the third batch is tested because deletion is complete

we could change the if calsuse below to check on this instead
```go
resp.DeletionProgress == v1.DeleteRelationshipsResponse_DELETION_PROGRESS_PARTIAL
```

then you can require for the partial case:
```go
require.Equal(len(beforeDelete)-len(afterDelete), 5)
```
and for the complete case
```go
require.Equal(len(beforeDelete)-len(afterDelete), 1)
```

Reply:
Until we change the test data and all of this has to be modified, vs the above which will work for any additional data added just by changing the expected-to-be-deleted map